thenewboston_node/business_logic/blockchain/base/network.py

```diff
@@ -67,3 +68,27 @@ def get_primary_validator(self, block_number: Optional[int] = None) -> Optional[
                 return self.get_node_by_identifier(account_number)
 
         return None
+
+    def get_node_role(self, identifier: hexstr) -> Optional[NodeRole]:
+        last_block_number = self.get_last_block_number()
+        node = self.get_node_by_identifier(identifier, last_block_number)
+
+        if node is None:
+            logger.warning(f'Getting node role of non-existent node: {identifier}')
+            return None
+
+        schedule = self.get_account_state_attribute_value(identifier, 'primary_validator_schedule', last_block_number)
+
+        if schedule is None:
+            return NodeRole.REGULAR_NODE
+
+        if schedule.is_block_number_in_future(last_block_number):
+            return NodeRole.REGULAR_NODE
+
+        if schedule.is_block_number_included(last_block_number):
+            return NodeRole.PRIMARY_VALIDATOR
+
+        if schedule.is_block_number_in_past(last_block_number):
+            return NodeRole.CONFIRMATION_VALIDATOR
+
+        return None
```

Comment:
is it logically possible to reach here ever?

if not, let's put an assert:
`assert False, 'Should never reach here'`

Reply:
instead the return

Reply:
@dmugtasimov initially i used "guard" expressions, but later found that lint force us to have return value in the end of the function, that's why I've chcanged code to nested expressions. On first review you asked to use "guard" expressions, so, i switched back to my first implementation with "guard" expressions.
Now, we need this return to make lint pass, otherwise we need to switch to nested expressions back.
What do you think?

Reply:
fix here: https://github.com/thenewboston-developers/thenewboston-node/pull/339